api/src/test/java/org/pmiops/workbench/api/BigQueryBaseTest.java

```diff
@@ -0,0 +1,107 @@
+package org.pmiops.workbench.api;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.cloud.bigquery.*;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.junit.After;
+import org.junit.Before;
+import org.pmiops.workbench.api.config.TestBigQueryConfig;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.lang.reflect.Type;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+@SpringBootTest(classes = {TestBigQueryConfig.class})
+public abstract class BigQueryBaseTest {
+
+    private static final Logger log = Logger.getLogger(BigQueryBaseTest.class.getName());
+
+    @Autowired
+    BigQuery bigquery;
+
+    @Autowired
+    WorkbenchConfig workbenchConfig;
+
+    public static final String BASE_PATH = "src/test/resources/bigquery/";
+
+    @Before
+    public void setUp() throws Exception {
+        createDataSet(workbenchConfig.bigquery.dataSetId);
+        for (String tableName: getTableNames()) {
+            createTable(workbenchConfig.bigquery.dataSetId, tableName);
+            insertData(workbenchConfig.bigquery.dataSetId, tableName);
+        }
+    }
+
+    @After
+    public void tearDown() {
+        for (String tableName: getTableNames()) {
+            deleteTable(workbenchConfig.bigquery.dataSetId, tableName);
+        }
+        deleteDataSet(workbenchConfig.bigquery.dataSetId);
+    }
+
+    public abstract List<String> getTableNames();
+
+    private void createDataSet(String dataSetId) {
+        Dataset dataSet = bigquery.create(DatasetInfo.newBuilder(dataSetId).build());
+    }
+
+    private void createTable(String dataSetId, String tableId) throws Exception {
+        ObjectMapper jackson = new ObjectMapper();
+        String rawJson =
+                new String(Files.readAllBytes(Paths.get(BASE_PATH + "schema/" + tableId.toLowerCase() + ".json")), Charset.defaultCharset());
+        JsonNode newJson = jackson.readTree(rawJson);
+
+        Gson gson = new Gson();
+        Schema schema = gson.fromJson(newJson.toString(), Schema.class);
+        StandardTableDefinition tableDef = StandardTableDefinition.of(schema);
+        bigquery.create(TableInfo.of(TableId.of(dataSetId, tableId), tableDef));
+    }
+
+    private boolean insertData(String dataSetId, String tableId) throws Exception {
```

Comment:
Rather than returning booleans from all of these, we should probably have them throw exceptions if anything fails. We don't want to have to check the return value everywhere I expect.

Reply:
throwing exceptions now.